src/neo4j_graphrag/experimental/pipeline/config/object_config.py

```diff
@@ -254,6 +254,10 @@ class ComponentConfig(ObjectConfig[Component]):
     DEFAULT_MODULE = "neo4j_graphrag.experimental.components"
     INTERFACE = Component
 
+    def get_run_params(self, resolved_data: dict[str, Any]) -> dict[str, Any]:
+        self._global_data = resolved_data
+        return self.resolve_params(self.run_params_)
+
 
 class ComponentType(RootModel):  # type: ignore[type-arg]
```

Comment:
Not part of this PR but do we know why this type: ignore comment is here?

Reply:
Yes, this is because `pydantic.RootModel` can be used either like this:
```
class ComponentType(RootModel):  # type: ignore[type-arg]
    root: Union[Component, ComponentConfig]
```
 or with a simple declaration like this:
```
ComponentType = RootModel[Union[Component, ComponentConfig]]
```
We use the former to add other methods to the class, but mypy is complaining that the type annotation `[]` is missing for RootModel.

Reply:
Hmm, that makes sense. Is there no way to point mypy to `root` in your example to resolve this issue? Feels a bit wrong to have this here. Maybe something we could look at in another PR in the future?

Reply:
I couldn't find a workaround but I have to admit I've not spent a lot of time on it. Will create a card so that we can investigate further.